src/theory/quantifiers/sygus/sygus_eval_unfold.cpp

```diff
@@ -158,7 +164,10 @@ void SygusEvalUnfold::registerModelValue(Node a,
               eval_children.end(), it->second[i].begin(), it->second[i].end());
           Node eval_fun = nm->mkNode(DT_SYGUS_EVAL, eval_children);
           eval_children.resize(1);
-          res = d_tds->unfold(eval_fun, vtm, exp);
+          // If we explicitly asked to unfold, we use single step, otherwise
+          // we use multi step.
+          res = unfold(eval_fun, vtm, exp, true, !do_unfold);
```

Comment:
So Booleans with anyconst will always be one-step unfolded? Why shouldn't anyconst subsume that? (I'm not sure it should, but this makes the case it definitely shouldn't).

Reply:
Why would anyconst subsume this?
If we are using symbolic constructors, we do the limited unfolding (traverse and unfold all but the any constant) in the case where we otherwise would do complete unfolding. This means that our policy on when the one step folding happens remains unchanged.

Reply:
The consequence of this is that we will never do full unfolding of Boolean operators, isn't it? Shouldn't we have this as an option?

Reply:
The option `options::sygusEvalUnfoldBool()` determines whether we do single-step or full unfolding of Bool operators. This is preserved by this PR.

Reply:
I see now.